python_modules/dagster/dagster/_core/definitions/op_invocation.py

```diff
@@ -433,7 +445,9 @@ def type_check_gen(gen):
 
 
 def _type_check_function_output(
-    op_def: "OpDefinition", result: T, context: "BoundOpExecutionContext"
+    op_def: "OpDefinition",
+    result: T,
+    context: Union["BoundOpExecutionContext", "BoundAssetExecutionContext"],
```

Comment:
I'm running into issues trying to consolidate this type hint. the function `validate_and_coerce_op_result_to_iterator` expects an `OpExecutionContext` so swapping this type hint to the `BoundExecutionContext` parent class causes type checker complaints